Add example for new generalized API (#2904)
timescale/docs: api/dimension_info.md

<Highlight type="tip">
For incompatible data types (for example, `jsonb`) you can specify a function to

> the `partition_func` to the dimension build which can extract a compatible
> data type. Look in the example section below.

Reviewer: ```suggestion
the `partition_func` argument of the dimension build to extract a compatible
data type. Look in the example section below.
```
